packages/dds/merge-tree/src/referencePositions.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Stack } from "./collections";
+import { ISegment } from "./mergeTree";
+import { ReferenceType, ICombiningOp } from "./ops";
+import { PropertySet, MapLike } from "./properties";
+
+export const reservedTileLabelsKey = "referenceTileLabels";
+export const reservedRangeLabelsKey = "referenceRangeLabels";
+
+export interface LabeledReference{
+    refType: ReferenceType,
+    properties?: PropertySet,
+}
+
+export const refGetTileLabels = (refPos: LabeledReference) =>
+    // eslint-disable-next-line no-bitwise
+    (refPos.refType & ReferenceType.Tile)
+        && refPos.properties ? refPos.properties[reservedTileLabelsKey] as string[] : undefined;
+
+export const refGetRangeLabels = (refPos: LabeledReference) =>
+    // eslint-disable-next-line no-bitwise
+    (refPos.refType & (ReferenceType.NestBegin | ReferenceType.NestEnd))
+        && refPos.properties ? refPos.properties[reservedRangeLabelsKey] as string[] : undefined;
+
+export function refHasTileLabel(refPos: LabeledReference, label: string) {
+    const tileLabels = refGetTileLabels(refPos);
+    if (tileLabels) {
+        for (const refLabel of tileLabels) {
+            if (label === refLabel) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+export function refHasRangeLabel(refPos: LabeledReference, label: string) {
+    const rangeLabels = refGetRangeLabels(refPos);
+    if (rangeLabels) {
+        for (const refLabel of rangeLabels) {
+            if (label === refLabel) {
+                return true;
+            }
+        }
+    }
+    return false;
+}
+export function refHasTileLabels(refPos: LabeledReference) {
+    return refGetTileLabels(refPos) !== undefined;
+}
+export function refHasRangeLabels(refPos: LabeledReference) {
+    return refGetRangeLabels(refPos) !== undefined;
+}
+
+export interface ReferencePosition {
+    properties?: PropertySet;
+    refType: ReferenceType;
```

Comment:
can remove these two and extend `LabeledReference`

Reply:
nice catch. i can just get rid of labeled reference and unify on reference position. labeled reference was an artifact of my refactoring that i missed removing.